profiling/src/string_table/borrowed.rs

```diff
@@ -0,0 +1,80 @@
+use ahash::RandomState;
+use std::collections::HashMap;
+use std::ops::Range;
+
+pub struct BorrowedStringTable<'a> {
+    pub vec: Vec<&'a str>,
+    pub map: HashMap<&'a str, usize, RandomState>,
```

Comment:
@morrisonlevi excuse my most likely super stupid question, but:

- `vec` holds references to `str` in the arena
- `map` uses the `&str` (the one also stored in `vec`) as key and the offset in `vec` as value

we do this because we do not want to / can't store pointers in the PHP_RUNTIME_CACHE, but offsets, correct?
Follow-up: why do we use a `usize`, If my assumption are correct, I would have defaulted to an `unsigned int`, just want to understand.

Reply:
Offsets in Rust are `usize`, so it's ergonomic to work with those. Serendipitously for this specific use case, the cache slot in the Zend Engine stores a pointer, and pointer-sized-integers are represented as `usize` in Rust. Make sense?